### PR TITLE
MCO-647: docs/machine-config-operator-certificates: Expand significantly

### PR DESCRIPTION
**Pull request by cgwalters**
Add a lot more details about what these certificates are, why they exist etc.

Version(s):
At least back to 4.11, and really most of this all the way back to 4.1.

Issue: https://issues.redhat.com/browse/MCO-642

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by cgwalters** [comment]
cc @yuqi-zhang

**Review by yuqi-zhang — APPROVED** [review]
/lgtm

Makes sense to me, thanks!

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc [hunk 1]
```diff
@@ -8,24 +8,43 @@ toc::[]
...
 
 .Additional resources
 
 * xref:../../networking/openshift_sdn/about-openshift-sdn.adoc#about-openshift-sdn[About the OpenShift SDN network plugin].
 
+=== Provisioning chain of trust
+
+The MCS CA is injected into the pointer configuration under the Ignition `security.tls.certificateAuthorities` configuration option. The Machine Config Server provides its "full" Ignition configuration via a webserver using the "MCS cert".
+
+Hence, this combination ensures that the client (Ignition running on a machine in the initramfs) is accessing the correct server because it validates that the MCS cert presented by the server has a chain of trust to an authority ("the MCS CA") it recognizes.
```
Comment (jldohmann) [inline security/certificate_types_descriptions/machine-config-operator-certificates.adoc:31]:
```suggestion
The MCS cert presented by the server must have a chain of trust to an authority, the MCS CA, it recognizes. This ensures that the client, Ignition running on a machine in the initramfs, is accessing the correct server.
```

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc [hunk 2]
```diff
@@ -8,24 +8,43 @@ toc::[]
...
+
+=== Key material inside a cluster
+
+The "MCS CA" appears in cluster as a configmap under the `kube-system` namespace, object `root-ca` and with key `ca.crt`.  The private key is not stored in cluster and is discarded after an installation is complete.
```
Comment (jldohmann) [inline security/certificate_types_descriptions/machine-config-operator-certificates.adoc:35]:
```suggestion
The MCS CA appears in the cluster as a config map under the `kube-system` namespace, object `root-ca` and with key `ca.crt`.  The private key is not stored in the cluster and is discarded after the installation completes.
```

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc [hunk 3]
```diff
@@ -8,24 +8,43 @@ toc::[]
...
+
+The "MCS cert" appears in cluster as a secret under the `openshift-machine-config-operator` namespace, object `machine-config-server-tls` with keys `tls.crt` and `tls.key`.
```
Comment (jldohmann) [inline security/certificate_types_descriptions/machine-config-operator-certificates.adoc:37]:
```suggestion
The MCS cert appears in the cluster as a secret under the `openshift-machine-config-operator` namespace, object `machine-config-server-tls` with keys `tls.crt` and `tls.key`.
```

**Review by jldohmann — COMMENTED** [review]
Preview link: https://60943--docspreview.netlify.app/openshift-enterprise/latest/security/certificate_types_descriptions/machine-config-operator-certificates.html

It look good! I left some feedback below. cc: @mburke5678

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc (hunk 1 above)
Comment (cgwalters) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:31]: Mmm...I think the new wording is wrong.  The MCS isn't recognizing the CA, the client is.
There may be more standard terms to use here.  Glancing at https://en.wikipedia.org/wiki/Transport_Layer_Security
we're talking about the phase here:

> The server usually then provides identification in the form of a [digital certificate](https://en.wikipedia.org/wiki/Public_key_certificate). The certificate contains the [server name](https://en.wikipedia.org/wiki/Hostname), the trusted [certificate authority](https://en.wikipedia.org/wiki/Certificate_authority) (CA) that vouches for the authenticity of the certificate, and the server's public encryption key.

The key (pun intended) thing here is that the client is verifying the server.

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc [hunk 4]
```diff
@@ -8,24 +8,43 @@ toc::[]
...
-Machine Config Operator certificates are used to secure connections between the Red Hat Enterprise Linux CoreOS (RHCOS) nodes and the Machine Config Server.
+This certificate authority is used to secure connections from nodes to Machine Config Server during initial provisioning.
+
+There are two certificates: a self-signed CA ("the MCS CA") as well as a derived certificate ("the MCS cert").
+
+=== Provisioning details
+
+OpenShift installations using {op-system-first} are installed using Ignition.  This Ignition configuration is split into two parts; the first part is a "pointer" config which is just a reference to a URL for the full configuration served by the Machine Config Server (MCS).
+
+The "pointer" Ignition config manifests as e.g. the `worker.ign` file created by `openshift-install` for User Provisioned Infrastucture installs. For Installer Provisioned Infrastructure (IPI) installs using the Machine API Operator, this configuration appears as the `worker-user-data` secret.
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:19]: Hmm, you think this part isn't useful?

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc (hunk 1 above)
Comment (jldohmann) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:31]: gotcha, I was iffy on this too. appreciate the clarification

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (cgwalters) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:19]: Oh I see, it's folded in above.

**Comment by cgwalters** [comment]
Thanks for reviewing!  I gave :+1: to the suggestions I agree with which is all of them except the TLS one.

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc (hunk 1 above)
Comment (jldohmann) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:31]:
```suggestion
The client, Ignition running on a machine in the initramfs, validates that the MCS cert presented by the server has a chain of trust to an authority, the MCS CA, it recognizes. This ensures that the client is accessing the correct server.
```

**Comment by mburke5678** [comment]
@cgwalters @jldohmann Would it make sense to also include this information in the proper MCO docs?

**Comment by cgwalters** [comment]
> @cgwalters @jldohmann Would it make sense to also include this information in the proper MCO docs?

It's tricky; I think we should just cross link for now.  The certificate documentation (here) is under "security and compliance" and it's something users who are focusing on that will be looking for, but not most use cases outside of it.

That said, this whole thing *does* start to document the intersection between the MCO and Machine API, which we could definitely do better at.

File: security/certificate_types_descriptions/machine-config-operator-certificates.adoc (hunk 1 above)
Comment (cgwalters) [reply security/certificate_types_descriptions/machine-config-operator-certificates.adoc:31]: LGTM

**Review by travier — APPROVED** [review]

**Pull request closed without merging**